### PR TITLE
Update MQTT 3.1.1 compliance text in doxygen and change log

### PR DESCRIPTION
**Pull request by aggarw13**
Update wording about MQTT library's compliance of the MQTT 3.1.1 specification in doxygen and CHANGELOG files﻿

### CODE REVIEW

**Review by abhidixi11 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,4 +3,4 @@
 ## v1.0.0 (September 2020)
 
 This is the first release of an MQTT client library in this repository.
-It has full compliance with the [MQTT 3.1.1 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349213).
\ No newline at end of file
+The MQTT library is a client-side implementation of the [MQTT 3.1.1 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349213), suitable for resource-constrained devices.
```
Comment (gkwicker) [inline CHANGELOG.md:6]: Eliminate the `#_Toc385...` at the end of the URL, this directs the user to deep within the page.

File: docs/doxygen/pages.txt [hunk 2]
```diff
@@ -54,7 +54,7 @@ The configurations for memory estimation are defined <a href="https://github.com
 @page mqtt_design Design
 @brief Architecture of the MQTT library.
 
-This MQTT client library provides a fully compliant implementation of the MQTT 3.1.1 spec. It is optimized for resource constrained devices and does not allocate any memory.
+This MQTT client library provides an implementation of the [MQTT 3.1.1 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349213). It is optimized for resource constrained devices and does not allocate any memory.
```
Comment (gkwicker) [inline docs/doxygen/pages.txt:57]: Eliminate the `#_Toc385...` at the end of the URL, this directs the user to deep within the page.

**Review by gkwicker — CHANGES_REQUESTED** [review]
See comments

File: docs/doxygen/pages.txt [hunk 3]
```diff
@@ -54,7 +54,7 @@ The configurations for memory estimation are defined <a href="https://github.com
 @page mqtt_design Design
```
Comment (yourslab) [inline docs/doxygen/pages.txt:54]: This file is now `pages.dox`

**Review by yourslab — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

**Pull request merged**